internal/keycloakb/mfa_tools.go

```diff
@@ -0,0 +1,41 @@
+package keycloakb
+
+import (
+	"context"
+
+	errorhandler "github.com/cloudtrust/common-service/errors"
+	kc "github.com/cloudtrust/keycloak-client"
+)
+
+var notMFATypes = map[string]bool{"password": true, "password-history": true}
+
+func CheckRemovableMFA(ctx context.Context, credentialID string, getCredentials func() ([]kc.CredentialRepresentation, error), logger Logger) error {
+	credentialsKc, err := getCredentials()
+	if err != nil {
+		logger.Warn(ctx, "err", err.Error())
+		return err
+	}
+	var found bool
+	var foundRemainingMFA bool
+	for _, credential := range credentialsKc {
+		if credential.ID != nil && *credential.ID != credentialID {
+			if value, ok := notMFATypes[*credential.Type]; !ok || !value {
```

Comment:
why do we need !ok || ! value ? !ok would not be enough ?

Reply:
In case someone would write `map[string]bool{"password": true, "password-history": true, "ctotp": false}`